ci/ci/build.py

```diff
@@ -88,22 +88,27 @@ def build(self, batch, code, scope):
             if step.scopes is None or scope in step.scopes:
                 step.build(batch, code, scope)
 
-        parents = set()
+        names_to_jobs = {}
         for step in self.steps:
-            parents.update(step.wrapped_job())
-        parents = list(parents)
+            wrapped = step.wrapped_job()
+            if wrapped:
+                names_to_jobs[step.name] = wrapped[0]
 
         if scope == 'dev':
             return
 
-        sink = batch.create_job('ubuntu:18.04',
-                                command=['/bin/true'],
-                                attributes={'name': 'sink'},
-                                parents=parents)
+        #Invert the step dependencies for cleanup:
+        cleanup_dependencies = {}
+        for step in self.steps:
+            for dep in step.deps:
+                if dep in cleanup_dependencies:
+                    cleanup_dependencies[dep].add(step.name)
+                else:
+                    cleanup_dependencies[dep] = {step.name}
 
         for step in self.steps:
             if step.scopes is None or scope in step.scopes:
-                step.cleanup(batch, scope, sink)
+                step.cleanup(batch, scope, [names_to_jobs[dep] for dep in cleanup_dependencies[step.name]])
```

Comment:
I'm worried about the case where the step doesn't end up in names_to_jobs because wrapped is None above.